app/src/main/java/org/fossasia/openevent/general/MainActivity.kt

```diff
@@ -111,4 +112,31 @@ class MainActivity : AppCompatActivity() {
                 .replace(R.id.frameContainer, fragment)
                 .commit()
     }
+
+    override fun onBackPressed() {
+        val currentFragmentClass = this.supportFragmentManager.findFragmentById(R.id.frameContainer).javaClass
+        when (currentFragmentClass) {
+            SearchFragment().javaClass -> {
+                loadFragment(EventsFragment())
+                supportActionBar?.title = "Events"
+                navigation.selectedItemId = navigation_events
```

Comment:
you are repeating the same code, think of some other way to acheive it

Reply:
Can I create a method for set event fragment?

Reply:
instead, you may use the logic `when currentFragmentClass is not EventFragment` since you want to get back to EventFragment in all case except current being EventFragment

Reply:
> instead, you may use the logic `when currentFragmentClass is not EventFragment` since you want to get back to EventFragment in all case except current being EventFragment

But if we want another condition in future like, need to load another fragment in a special case then it was a problem there. Should I change it?

Reply:
then it would be changed accordingly for now just change it :)